src/tests/multihost/ad/test_dyndns.py

```diff
@@ -0,0 +1,312 @@
+          7. Wait for SSSD to update the IP, refresh interval + update timeout
+          8. Perform DNS lookup for hostname on dummy interface
+          9. Perform DNS lookup for IP on dummy interface
+        :expectedresults:
+          1. Client joins domain
+          2. Should succeed
+          3. Should succeed
+          4. Hostname resolves correctly on dummy interface
+          5. IP resolves correctly on dummy interface
+          6. New IP address is assigned for dummy interface
+          7. Wait for some time
+          8. Hostname resolves correctly on dummy interface with new IP
+          9. IP resolves correctly on dummy interface with new hostname
+        """
+        (dummy_interface, dummy_ip) = extra_interface
+        dummy_ip_after_refresh = '192.168.1.' + str(random.randrange(2, 255))
+        while dummy_ip == dummy_ip_after_refresh:
+            dummy_ip_after_refresh = '192.168.1.' + str(random.randrange(2, 255))
+        update_timeout = 20
+        refresh_interval = 61
+        client = sssdTools(multihost.client[0], multihost.ad[0])
+        hostname_cmd = multihost.client[0].run_command('hostname', raiseonerr=False)
+        hostname = hostname_cmd.stdout_text.rstrip()
+        domain = multihost.ad[0].domainname
+
+        adjoin(membersw='adcli')
+        domain_section = 'domain/{}'.format(domain)
+        sssd_params = {
+            'dyndns_iface': dummy_interface,
+            'dyndns_refresh_interval': refresh_interval}
+        client.sssd_conf(domain_section, sssd_params)
+        client.clear_sssd_cache()
+
+        time.sleep(update_timeout)
+        cmd_lkup = multihost.client[0].run_command(f"nslookup {hostname}", raiseonerr=False)
+        cmd_ptr_lkup = multihost.client[0].run_command(f"nslookup {dummy_ip}", raiseonerr=False)
+        assert cmd_lkup.returncode == 0
+        assert cmd_ptr_lkup.returncode == 0
+        multihost.client[0].run_command('ip addr flush dev ' + dummy_interface)
+        multihost.client[0].run_command('ip addr add ' + dummy_ip_after_refresh + ' dev ' + dummy_interface)
+
+        time.sleep(update_timeout)
+        cmd_forward_lookup = multihost.client[0].run_command(f"nslookup {hostname}")
+        cmd_ptr_lookup = multihost.client[0].run_command(f"nslookup {dummy_ip}")
+        assert cmd_forward_lookup.returncode == 0
+        assert cmd_ptr_lookup.returncode == 0
+        multihost.client[0].run_command('ip addr flush dev ' + dummy_interface)
+        multihost.client[0].run_command('ip addr add ' + dummy_ip_after_refresh + ' dev ' + dummy_interface)
+
+        time.sleep(refresh_interval)
+        cmd_forward_lookup = multihost.client[0].run_command(f"nslookup {hostname}")
+        cmd_ptr_lookup = multihost.client[0].run_command(f"nslookup {dummy_ip_after_refresh}")
+        assert cmd_forward_lookup.returncode == 0
+        assert cmd_ptr_lookup.returncode == 0
+
+    @classmethod
+    def class_teardown(cls, multihost):
```

Comment:
The class_teardown does not seem to be needed, there is missing LF at the end of the file.

Reply:
I fixed it in another PR.  https://github.com/SSSD/sssd/pull/6348

Reply:
The PR #6348 is closed without merging and this class_teardown still has no code.